cupyx/scipy/interpolate/_rgi.py

```diff
@@ -0,0 +1,347 @@
+                                          spacing (suitable for e.g., N-D image
+                                          resampling)
+
+    """
+    # sanity check 'method' kwarg
+    if method not in ["linear", "nearest"]:
+        raise ValueError(
+            "interpn only understands the methods 'linear', "
+            "'nearest'. 'splinef2d' is not yet implemented. "
+            "You provided %s." % method)
+
+    if not hasattr(values, 'ndim'):
+        values = cp.asarray(values)
+
+    ndim = values.ndim
+
+    # sanity check consistency of input dimensions
+    if len(points) > ndim:
+        raise ValueError("There are %d point arrays, but values has %d "
+                         "dimensions" % (len(points), ndim))
+
+    points = list(points)
+    # sanity check input grid
+    for i, p in enumerate(points):
+        diff_p = cp.diff(p)
+        if not cp.all(diff_p > 0.):
+            if cp.all(diff_p < 0.):
+                # input is descending, so make it ascending
+                points[i] = points[i][::-1]
+                values = cp.flip(values, axis=i)
+            else:
+                raise ValueError("The points in dimension %d must be strictly "
+                                 "ascending or descending" % i)
+        if not cp.asarray(p).ndim == 1:
+            raise ValueError("The points in dimension %d must be "
+                             "1-dimensional" % i)
+        if not values.shape[i] == len(p):
+            raise ValueError("There are %d points and %d values in "
+                             "dimension %d" % (len(p), values.shape[i], i))
+    grid = tuple([cp.asarray(p) for p in points])
+
+    # sanity check requested xi
+    xi = _ndim_coords_from_arrays(xi, ndim=len(grid))
+    if xi.shape[-1] != len(grid):
+        raise ValueError("The requested sample points xi have dimension "
+                         "%d, but this RegularGridInterpolator has "
+                         "dimension %d" % (xi.shape[-1], len(grid)))
+
+    if bounds_error:
+        for i, p in enumerate(xi.T):
+            if not cp.logical_and(cp.all(grid[i][0] <= p),
+                                  cp.all(p <= grid[i][-1])):
+                raise ValueError("One of the requested xi is out of bounds "
+                                 "in dimension %d" % i)
+
+    # perform interpolation
+    if method == "linear":
+        interp = RegularGridInterpolator(points, values, method="linear",
+                                         bounds_error=bounds_error,
+                                         fill_value=fill_value)
+        return interp(xi)
+    elif method == "nearest":
+        interp = RegularGridInterpolator(points, values, method="nearest",
+                                         bounds_error=bounds_error,
+                                         fill_value=fill_value)
+        return interp(xi)
+    elif method == "splinef2d":
+        raise NotImplementedError("splinef2d not implemented yet")
```

Comment:
This I think can be removed: this method in scipy dates to before `method="cubic"` was a thing and is no longer very even in scipy, let alone in CuPy going forward.

Reply:
Thanks!  Removed.